src/main/scala/com/earldouglas/xwp/ContainerPlugin.scala

```diff
@@ -0,0 +1,136 @@
+package com.earldouglas.xwp
+
+import java.io.File
+import java.util.concurrent.atomic.AtomicReference
+
+import sbt._, Keys._
+
+object ContainerPlugin extends AutoPlugin {
+
+  lazy val start = taskKey[Process]("start container")
+  lazy val stop  = taskKey[Unit]("stop container")
+
+  object autoImport {
+    val Container = config("container").hide
+
+    val containerLibs        = settingKey[Seq[ModuleID]]("container libraries")
+    val containerMain        = settingKey[String]("container main class")
+    val containerPort        = settingKey[Int]("port number to be used by container")
+    val containerConfigFile  = settingKey[Option[File]]("path of container configuration file")
+    val containerArgs        = settingKey[Seq[String]]("additional container args")
+    val containerLaunchCmd   = settingKey[Seq[String]]("command to launch container")
+    val containerForkOptions = settingKey[ForkOptions]("fork options")
+  }
+
+  private lazy val containerInstance = settingKey[AtomicReference[Option[Process]]]("Current container process")
```

Comment:
I'm not 100% confident of this, but tests pass either on my local machine and travis.